src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -408,16 +407,18 @@ private void createImage(JlinkConfiguration config) throws Exception {
         }
 
         // First create the image provider
-        ImageProvider imageProvider = createImageProvider(config,
-                                                          options.packagedModulesPath,
-                                                          options.ignoreSigning,
-                                                          options.bindServices,
-                                                          options.verbose,
-                                                          log);
+        ImageHelper imageProvider = createImageProvider(config,
```

Comment:
With the updated change, `createImageProvider` can continue to return `ImageProvider`.   This change can be reverted.